File: projects/heart-linker-app/android/app/src/main/java/com/meishi/app/model/SnsType.kt
package com.meishi.app.model

/**
 * SNSの種類。typeKeyはDB保存用、code はMessagePackスキーマのservice_type enum値。
 */
enum class SnsType(val typeKey: String, val displayName: String, val code: Int) {
    TWITTER("twitter", "X (Twitter)", 1),
    FACEBOOK("facebook", "Facebook", 2),
    LINE("line", "LINE", 3),
    WHATSAPP("whatsapp", "WhatsApp", 4),
    INSTAGRAM("instagram", "Instagram", 5),

    // 「それ以外」として下部に表示する、それらしい名前の候補
    TIKTOK("tiktok", "TikTok", 6),
    YOUTUBE("youtube", "YouTube", 7),
    GITHUB("github", "GitHub", 8),
    LINKEDIN("linkedin", "LinkedIn", 9),
    WEBSITE("website", "Webサイト・その他URL", 10),
    OTHER("other", "その他", 0);

    companion object {
        val primary = listOf(TWITTER, FACEBOOK, LINE, WHATSAPP, INSTAGRAM)
        val others = listOf(TIKTOK, YOUTUBE, GITHUB, LINKEDIN, WEBSITE, OTHER)

        fun fromKey(key: String): SnsType =
            values().firstOrNull { it.typeKey == key } ?: OTHER

        fun fromCode(code: Int): SnsType =
            values().firstOrNull { it.code == code } ?: OTHER
    }
}

File: projects/heart-linker-app/ios/Meishi/Models/SnsType.swift
import Foundation

/// SNS種別。code は MessagePack の service_type enum 値と対応する。
enum SnsType: String, CaseIterable, Identifiable {
    case twitter, facebook, line, whatsapp, instagram
    case tiktok, youtube, github, linkedin, website, other

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .twitter:   return "X (Twitter)"
        case .facebook:  return "Facebook"
        case .line:      return "LINE"
        case .whatsapp:  return "WhatsApp"
        case .instagram: return "Instagram"
        case .tiktok:    return "TikTok"
        case .youtube:   return "YouTube"
        case .github:    return "GitHub"
        case .linkedin:  return "LinkedIn"
        case .website:   return "Webサイト・その他URL"
        case .other:     return "その他"
        }
    }

    /// SF Symbols アイコン名
    var symbolName: String {
        switch self {
        case .twitter:   return "at"
        case .facebook:  return "globe"
        case .line:      return "bubble.left"
        case .whatsapp:  return "bubble.left.fill"
        case .instagram: return "camera"
        case .tiktok:    return "music.note"
        case .youtube:   return "play.rectangle"
        case .github:    return "chevron.left.forwardslash.chevron.right"
        case .linkedin:  return "briefcase"
        case .website:   return "link"
        case .other:     return "ellipsis.circle"
        }
    }

    var code: Int {
        switch self {
        case .other:     return 0
        case .twitter:   return 1
        case .facebook:  return 2
        case .line:      return 3
        case .whatsapp:  return 4
        case .instagram: return 5
        case .tiktok:    return 6
        case .youtube:   return 7
        case .github:    return 8
        case .linkedin:  return 9
        case .website:   return 10
        }
    }

    static func fromCode(_ code: Int) -> SnsType {
        allCases.first { $0.code == code } ?? .other
    }

    static let primary: [SnsType] = [.twitter, .facebook, .line, .whatsapp, .instagram]
    static let others:  [SnsType] = [.tiktok, .youtube, .github, .linkedin, .website, .other]
}
